### ISSUE TITLE
Add an "experimental" flag for rules

### ISSUE THREAD

**Issue by rgeyer** [grafana/dashboard-linter#185]
New rules, particularly ones which replace deprecated rules (see #184) should be introduced with an "experimental" flag.

Experimental rules would not be added/run automatically

When many experimental rules exist at the same time, individual rules can be enabled in the `.lint` configuration file (mechanism to be determined).

*All* currently experimental rules can be enabled with a CLI flag. This would allow users to "dry run" the latest changes against their dashboards/repositories.

When an experimental rule becomes part of the standard ruleset, the `.lint` configuration should be checked for tidiness. I.E. Formerly experimental rules that are still "enabled" in the config can be removed, the user should receive a warning/text output on on STDOUT.

**Comment by alex5517** [comment]
@rgeyer,

As for selectively enabling experimental rules, instead of enabling all rules by providing an "experimental" flag.
Would we not be able to "avoid" the config check, warning text etc. if we turn it around and let it work the same way as the "stable" rules does?

So if one would want to use 2 out of 3 experimental rules, then they use the "experimental" flag and then create an exclusion for the experimental rule they do not want?

**Comment by rgeyer** [comment]
> @rgeyer,
> 
> As for selectively enabling experimental rules, instead of enabling all rules by providing an "experimental" flag. Would we not be able to "avoid" the config check, warning text etc. if we turn it around and let it work the same way as the "stable" rules does?
> 
> So if one would want to use 2 out of 3 experimental rules, then they use the "experimental" flag and then create an exclusion for the experimental rule they do not want?

Right, excellent point. I wrote this up in a hurry and didn't think through it. I think your solution makes the most sense.

I'll update the issue accordingly. Thanks!
